### ISSUE TITLE
Position of functional actions overwritten if selection is enabled

### ISSUE THREAD

**Issue by TheGoodSpinach** [material-table-core/core#655]
**Describe the bug**
This is a repeat of #153, since I can't seem to re-open that issue. The bug still seems to present as of v4.3.44.

When attempting to use a function action with a set position, e.g.
```
actions={[rowData => {
       position: "row"
}]}
```

The position is not respected if `selection` is set to `true` on the table, instead forcing the `position` to `toolbarOnSelect`

**To Reproduce**

https://codesandbox.io/embed/material-table-starter-template-forked-6kc8vz?fontsize=14&hidenavigation=1&theme=dark

1. This table has two actions, "Save User" which should be on the row, and "Save Users" which should be on toolbar on selection. Observe that neither action is present initially
2. Select a row
3. Both actions present in toolbar on selection.

**Expected behavior**
Action with position "row" should be present on the row, alongside the checkbox.

**Comment by TheGoodSpinach** [comment]
I believe the issue is here:

https://github.com/material-table-core/core/blob/d8fb7db073cd2b2f55c05289816dc2f8bcd168db/src/material-table.js#L314-L316

This probably should evaluate `action` to determine the correct position, and default to `toolbarOnSelect` only if none is provided.

**Comment by TheGoodSpinach** [comment]
So it appears I was not understanding the solution implemented in #154. Doing this _does_ work:

```
actions={[{
     action: rowData => {
         icon: "foo",
         onClick: () => {}
     },
     position: "row"
}]}
```

However, I wasn't able to find this implementation documented anywhere, and in my opinion it is pretty unintuitive. Is it worth considering still fixing the bug to allow this implementation to work?

```
actions={[ rowData => ({
         icon: "foo",
         onClick: () => {},
         position: "row"
     })
]}
```
